kotlin/spatial-operations/src/main/java/com/esri/arcgisruntime/sample/spatialoperations/MainActivity.kt

```diff
@@ -0,0 +1,209 @@
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  private val inputGeometryOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+  private val resultGeometryOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+
+  // simple black (0xFF000000) line symbol for outlines
+  private val lineSymbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, -0x1000000, 1f)
+  private val resultFillSymbol = SimpleFillSymbol(
+    SimpleFillSymbol.Style.SOLID, -0x16e0e1, lineSymbol
+  )
+  private var inputPolygon1: Polygon? = null
+  private var inputPolygon2: Polygon? = null
+
+  // The spatial operation switching menu items.
+  private var noOperationMenuItem: MenuItem? = null
+  private var intersectionMenuItem: MenuItem? = null
+  private var unionMenuItem: MenuItem? = null
+  private var differenceMenuItem: MenuItem? = null
+  private var symmetricDifferenceMenuItem: MenuItem? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create ArcGISMap with topographic basemap
+    ArcGISMap(Basemap.createLightGrayCanvas()).let {
+      mapView.map = it
+    }
+
+    // create graphics overlays to show the inputs and results of the spatial operation
+    mapView.graphicsOverlays.add(inputGeometryOverlay)
+    mapView.graphicsOverlays.add(resultGeometryOverlay)
+
+    // create input polygons and add graphics to display these polygons in an overlay
+    createPolygons()
+
+    GeometryEngine.union(inputPolygon1, inputPolygon2).extent.let {
+      mapView.setViewpointGeometryAsync(it, 20.0)
+    }
+  }
+
+  private fun showGeometry(resultGeometry: Geometry) {
+    // add a graphic from the result geometry, showing result in red (0xFFE91F1F)
+    Graphic(resultGeometry, resultFillSymbol).let {
+      resultGeometryOverlay.graphics.add(it)
+
+      // select the result to highlight it
+      it.isSelected = true
+    }
+  }
+
+  private fun createPolygons() {
+    // create input polygon 1
+    PointCollection(SpatialReferences.getWebMercator()).run {
+      add(Point(-13160.0, 6710100.0))
+      add(Point(-13300.0, 6710500.0))
+      add(Point(-13760.0, 6710730.0))
+      add(Point(-14660.0, 6710000.0))
+      add(Point(-13960.0, 6709400.0))
+      inputPolygon1 = Polygon(this)
+    }
+
+    // create and add a blue graphic to show input polygon 1
+    SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, -0x66ffff34, lineSymbol).let {
+      inputGeometryOverlay.getGraphics().add(Graphic(inputPolygon1, it))
+    }
+
+    // outer ring
+    val outerRing = Part(PointCollection(SpatialReferences.getWebMercator()).also {
```

Comment:
Also is generally used to do stuff in addition to the creation of an object like

.also {
Log.d(TAG, "This is called: " + it.name
}

Let's use apply. { ... } here